components/cards/Status.tsx

```diff
@@ -1,74 +1,142 @@
+        if (from < now) {
+          setFromTime(now)
+          setToTime(new Date(now.getTime() + 60 * 60 * 1000))
+        } else {
+          setFromTime(from)
+          setToTime(to)
+        }
+      } else {
+        const now = new Date()
+        setFromTime(now)
+        setToTime(new Date(now.getTime() + 60 * 60 * 1000))
+      }
+    }
+
+    loadSavedTimes()
+  }, [])
+
+  const formatTime = (date: Date) => {
+    const hours = date.getHours()
+    const minutes = date.getMinutes()
+    const formattedHours = hours % 12 || 12
+    const formattedMinutes = minutes < 10 ? `0${minutes}` : minutes
+    return `${formattedHours}:${formattedMinutes}`
+  }
+
+  const saveTimes = async (from: Date, to: Date) => {
+    const now = new Date()
+    const validFrom = from < now ? now : from
+    let validTo =
+      to > validFrom ? to : new Date(validFrom.getTime() + 60 * 60 * 1000)
+    const maxTo = new Date(validFrom.getTime() + 24 * 60 * 60 * 1000)
+    if (validTo > maxTo) validTo = maxTo
+
+    setFromTime(validFrom)
+    setToTime(validTo)
+
+    await AsyncStorage.setItem("fromTime", validFrom.toISOString())
+    await AsyncStorage.setItem("toTime", validTo.toISOString())
+    const timeRangeDisplay = `${formatTime(validFrom)}-${formatTime(validTo)}`
 
-  const handleTimeSlotChange = (selectedTime: string) => {
     setTemporaryStatus((prev: TemporaryStatusType) => ({
       ...prev,
-      timeSlot: selectedTime,
+      timeSlot: timeRangeDisplay,
     }))
   }
 
+  const handleTimeSlotChange = (selectedTime: string) => {
+    if (selectedTime.toLowerCase() === "set time") {
+      setIsDatePickerOpen(true)
+    } else {
+      setTemporaryStatus((prev: TemporaryStatusType) => ({
+        ...prev,
+        timeSlot: selectedTime,
+      }))
+    }
+  }
+
   return (
     <View style={styles.container}>
-      <CustomTitle text="When" style={styles.title} />
-      <View style={styles.scrollContainer}>
-        <ScrollView
-          horizontal={true}
-          showsHorizontalScrollIndicator={false}
-          contentContainerStyle={styles.scrollContentContainer}>
-          <View style={styles.buttonContainer}>
-            {timeSlots.map((slot) => (
-              <TouchableOpacity
-                onPress={() => handleTimeSlotChange(slot)}
-                key={slot}>
-                <Badge
-                  name={slot}
-                  variant={
-                    temporaryStatus.timeSlot.toLowerCase() ===
-                    slot.toLowerCase()
-                      ? "default"
-                      : "outline"
-                  }
-                  style={styles.badge}
-                />
-              </TouchableOpacity>
-            ))}
+      {isDatePickerOpen ? (
+        <DatePicker
+          onClose={() => setIsDatePickerOpen(false)}
+          initialFromTime={fromTime ?? new Date()}
+          initialToTime={
+            toTime ?? new Date(new Date().getTime() + 60 * 60 * 1000)
+          }
```

Comment:
You can also abstract these two props, instead of passing them from outside ; they call all live in the component